### ISSUE TITLE
Saucelab tests are failing on the master branch

### ISSUE THREAD

**Issue by 36degrees** [alphagov/accessible-autocomplete#394]
The Saucelab tests are currently failing on the master branch:

```
element click intercepted: Element <strong>...</strong> is not clickable at point (263, 362).
Other element would receive the click: <li aria-selected="false" class="autocomplete__option" id="autocomplete-customTemplates__option--0" role="option" tabindex="-1" aria-posinset="1" aria-setsize="1">...</li>
(Session info: chrome=78.0.3904.70)\n (Driver info: chromedriver=78.0.3904.70 (edb9c9f3de0247fd912a77b7f6cae7447f6d3ad5-refs/branch-heads/3904@{#800}),platform=Windows NT 10.0.10586 x86_64)
```

They appear to have been failing since #393 was merged.

**Comment by peterjaric** [comment]
Sorry about this, I overlooked the Sauce Labs section in CONTRIBUTING.md.

**Comment by 36degrees** [comment]
@peterjaric please don't worry about it – it's a known issue that because SauceLabs requires an API key, the tests don't run on contributions from external contributors. Really we need to improve our processes to try and catch this stuff earlier, but in the meantime we expect this to come up occasionally.

Thank you again for your contribution – we really appreciate you helping to improve the accessible autocomplete 🙂

**Comment by peterjaric** [comment]
OK, I'll let you handle it then! 

(Also since I got this error message after creating a free trial account at Sauce Labs and running the tests: "ERROR: Uh oh, you've run out of minutes!" 😄 )

**Comment by nickcolley** [comment]
Just investigating this, I don't think that pull request was the cause of these tests failing, it seems to be unrelated.

**Comment by nickcolley** [comment]
I was curious if we're just seeing general flakeyness in the tests so I triggered a rebuild of master tests.

```
1) Accessible Autocomplete basic example should announce status changes using two alternately updated aria live regions:
expected the first aria live region to be populated within 10000 milliseconds
running chrome on Windows 10
Error: expected the first aria live region to be populated within 10000 milliseconds
    at elementIdText("0.5907962755285172-4") - getText.js:35:50
```

Which failed on a known flakey test. Going to try once more and see what happens.

**Comment by nickcolley** [comment]
Hmm, this time Travis completes and it looks like everything is fine but the Saucelabs dashboard reports failures? 🤔

**Comment by nickcolley** [comment]
OK, so I think it is a problem introduced by that pull request, the tests rely on a specific ordering currently so the new test broke this assumption.

**Comment by nickcolley** [comment]
What I've noticed is that Travis will return a failed build if the assertions in the codebase do not return as expected. However, Saucelabs will show a failure for flakey tests runs or intentionally failing tests.

With this in mind I think we should remove or ignore the Saucelabs dashboard matrix and instead focus on the Travis build.

**Comment by 36degrees** [comment]
I checked out an older commit from October (2010e713b6f53ad8a715c52ccc3e34dda1daace3) and ran `npm run wdio` and the builds passed in Saucelabs (apart from Chrome, but that's been failing for a while). Using `git bisect` it looks like `83d79858bc3d9c190c2b74d80cbc50b9ca955ddc` is the commit that causes the failures.

As far as I can tell, the `wdio-sauce-service` is what is responsible for the 'status' that shows up in SauceLabs – SauceLabs by itself doesn't really have any notion of pass/fail for the test suite as a whole. Although it does flag events that fail in the event list, these don't seem to relate to the overall status, as there are tests that are shown as passed that have these too.

There seems to be a discrepancy between the status that gets reported back by the `wdio-sauce-service` (to SauceLabs) and the status that is shown by the reporter (which is what we see in Travis). Adding logging in to the wdio-sauce-service afterTest hook shows that the `test.passed` property is `false` for 'should allow typing' test. This appears to be a legitimate failure, but I've no idea why it's not showing up in the reporter / Travis.

The reason the test is failing is that the autocomplete is not being reset correctly between tests – at the point the new aria-selected test finishes running the autocomplete menu is still open, and so clearing the autocomplete input doesn't correctly reset it. I've opened a PR to close open autocompletes by sending the Esc key (#401) which seems to fix the failing tests at least.
